packages/core/src/services/scales-cartesian.ts

```diff
@@ -118,6 +142,64 @@ export class CartesianScales extends Service {
 		}
 	}
 
+	isDualAxes() {
+		return this.isDualAxes;
+	}
+
+	// if any of the axes objects have datasets [] asserted we flag the chart as dual axes
+	// it does not count as dual axes if it just has another axis turned on but is not actually using it to map a dataset
+	determineDualAxes() {
+		const options = this.model.getOptions();
+		const axesOptions = Tools.getProperty(options, "axes");
+
+		if (axesOptions[AxisPositions.LEFT]?.datasets  &&  axesOptions[AxisPositions.RIGHT]
+			|| axesOptions[AxisPositions.RIGHT]?.datasets  &&  axesOptions[AxisPositions.LEFT]
+			|| axesOptions[AxisPositions.TOP]?.datasets  &&  axesOptions[AxisPositions.BOTTOM]
+			|| axesOptions[AxisPositions.BOTTOM]?.datasets  &&  axesOptions[AxisPositions.TOP]) {
+				this.dualAxes = true;
```

Comment:
is it `this.dualAxes` or `this.isDualAxes`?

Also, what about `determineAxisDuality` for the name?